.github/workflows/e2e-tests-manual.yaml

```diff
@@ -0,0 +1,50 @@
+# Used to run the E2E tests against a branch of your choice from the github.com UI.
```

Comment:
Is it possible to merge this file and e2e-tests-scheduled by having both `workflow_dispatch` and `cron` under `on`?

Reply:
The job is different between the two. `-manual` has an input for the branch whereas `-scheduled` has a matrix dimension.

Reply:
Could BRANCH be set with an if? Something like "if github.events.input.branch is set, use it; else use main"?

Reply:
I'm not talking about the env var. I'm talking about the matrix dimension.

https://github.com/Azure/iot-identity-service/blob/995f6306b3c45a723af6b215b2c1a67126286db1/.github/workflows/e2e-tests-scheduled.yaml#L23-L24

Reply:
```
branch:
- 'main'
  if github.event_name = ...
```
Haven't tried it, but would that work?

Reply:
https://docs.github.com/en/free-pro-team@latest/actions/reference/workflow-syntax-for-github-actions

Reply:
Yeah, I wasn't paying attention to the correct syntax.

Point is: I don't think we should have two almost identical files that require manual updates to both (`DEVNOTE: Keep in sync with ...`) because when making changes to one of them, it's easy to forget to update the other. So combining them into one file would be more maintainable in the future. I'm not familiar with GitHub Actions (haven't used it before) but I'd imagine there's some way to do this.

Reply:
Sure. If you find some way to do it, please tell me. I did this after already having checked, but of course I could've missed something.

Reply:
I'll take a look, maybe in a few days. I'll sign off for now.